ui/ozone/platform/wayland/wayland_connection.cc

```diff
@@ -325,6 +325,31 @@ bool WaylandConnection::IsSelectionOwner() {
   return !!data_source_;
 }
 
+void WaylandConnection::RequestDragData(const std::string& mime_type) {
+  data_device_->RequestDragData(mime_type);
+}
+
+void WaylandConnection::StartDrag(std::vector<std::string>& mime_types,
+                                  int operation,
+                                  const SkBitmap& bitmap) {
+  if (!data_source_) {
+    wl_data_source* data_source = data_device_manager_->CreateSource();
+    drag_data_source_.reset(new WaylandDataSource(data_source));
```

Comment:
std::make_unique\<type\>

Reply:
Btw, can you move creation of the WaylandDataSource into a separate method and share it with the OfferClipboardData? Maybe, set_connection can also be removed and WaylandConnection can be passed as an argument in the construction parameters.